common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/ShuffleTransportContextSuite.java

```diff
@@ -90,15 +94,22 @@ private ByteBuf getDecodableMessageBuf(Message req) throws Exception {
   public void testInitializePipeline() throws IOException {
     // SPARK-43987: test that the FinalizedHandler is added to the pipeline only when configured
     for (boolean enabled : new boolean[]{true, false}) {
-      ShuffleTransportContext ctx = createShuffleTransportContext(enabled);
-      SocketChannel channel = new NioSocketChannel();
-      RpcHandler rpcHandler = mock(RpcHandler.class);
-      ctx.initializePipeline(channel, rpcHandler);
-      String handlerName = ShuffleTransportContext.FinalizedHandler.HANDLER_NAME;
-      if (enabled) {
-        Assertions.assertNotNull(channel.pipeline().get(handlerName));
-      } else {
-        Assertions.assertNull(channel.pipeline().get(handlerName));
+      for (boolean isClient: new boolean[]{true, false}) {
+        // Since the decoder is not Shareable, reset it between test runs to avoid errors since it's
+        // used both across ShuffleTransportContextSuite and SslShuffleTransportContextSuite
+        // and server/clients
```

Comment:
The decoder is not being used here (other than configuring the pipeline) - why do we need to reset it ?

Reply:
Why not tag it as `@Sharable` ? It is stateless anyway, right ?

Reply:
My general bias was to not add a constraint on production code just for tests -- but it makes sense to tag this as shareable, so I'll do that.

Reply:
Completely agree, I would prefer not modifying prod code for tests (even though we do have a fair bit of `Utils.isTesting` in our code :-( ).
In this case, it is orthogonal though - since it actually makes sense as well - it was never done before due to oversight, not due to anything else I guess !